### ISSUE TITLE
WHOLE-INDEX not detected in include file

### ISSUE THREAD

**Issue by maelstraggiotti** [Riverside-Software/sonar-openedge#1189]
The server and SonarLint do not detect a `whole-index` search within a `for each` statement when it is located in an include file, even though the XREF is correct and parsed by the analyzer.

The statement is correctly flagged when it is written directly in the program.

---
OE 12.8.6
CABL 3.1
Sonarlint 11.7.4801
---

Procedure
```
{incl.i}
```

Include
```
FOR EACH Table
    WHERE Table.SecondIndexField EQ 1
    NO-LOCK :
    /* Noop */
END.
```

Xref
```
   <Reference Reference-type="ACCESS" Object-identifier="SecondIndexField">
      <Source-guid>1uCAoRrYoJDcFLHKEDl+CA</Source-guid>
      <File-num>2</File-num>
      <Ref-seq>7</Ref-seq>
      <Line-num>1</Line-num>
      <Object-context>Db.Table</Object-context>
      <Access-mode/>
      <Data-member-ref/>
      <Temp-ref/>
      <Detail/>
      <Is-static>false</Is-static>
      <Is-abstract>false</Is-abstract>
    </Reference>
    <Reference Reference-type="SEARCH" Object-identifier="Db.Table">
      <Source-guid>1uCAoRrYoJDcFLHKEDl+CA</Source-guid>
      <File-num>2</File-num>
      <Ref-seq>8</Ref-seq>
      <Line-num>1</Line-num>
      <Object-context>Index0</Object-context>
      <Access-mode/>
      <Data-member-ref/>
      <Temp-ref/>
      <Detail>WHOLE-INDEX</Detail>
      <Is-static>false</Is-static>
      <Is-abstract>false</Is-abstract>
    </Reference>
```

**Comment by maelstraggiotti** [comment]
The repro case is indeed correctly detected by the scanner. I understand it is not with SonarLint. The actual case is actually not detected at all, I'll send you the XREF.

**Comment by gquerret** [comment]
As seen offline, not able to reproduce the issue.
